### PR TITLE
[STOBJECT] Add support for the mouse key tray icon

### PR DESCRIPTION
**Pull request by learn-more**
Additionally:
* [PSDK] Add missing defines / argument names
* [PSDK] Add COWAIT_DEFAULT
* [STOBJECT] Keep the object alive while the thread is running,
  * Fix shutting down the thread.

And because github can't really preview icons:

image

### CODE REVIEW

File: dll/shellext/stobject/csystray.cpp [hunk 1]
```diff
@@ -269,8 +282,11 @@ HRESULT CSysTray::CreateSysTrayThread()
 
 HRESULT CSysTray::DestroySysTrayWindow()
 {
-    DestroyWindow();
-    hwndSysTray = NULL;
+    if (!DestroyWindow())
+    {
+        // Window is from another thread, ask it politely to destroy itself:
```
Comment (HBelusca) [inline dll/shellext/stobject/csystray.cpp:287]:
```suggestion
        // Window is from another thread, ask it politely to destroy itself.
```

**Comment by HBelusca** [comment]
The light gray in the icon is not quite evidently visible (it could be mistaken with white, if one doesn't pay close attention to the icon)

**Comment by learn-more** [comment]
> The light gray in the icon is not quite evidently visible (it could be mistaken with white, if one doesn't pay close attention to the icon)

How about this?

image

**Comment by HBelusca** [comment]
> How about this?

It's now definitively more visible :D

File: dll/shellext/stobject/csystray.cpp [hunk 2]
```diff
@@ -132,12 +141,14 @@ HRESULT CSysTray::ShutdownIcons()
     {
         if (this->dwServicesEnabled & g_IconHandlers[i].dwServiceFlag)
         {
+            this->dwServicesEnabled &= ~g_IconHandlers[i].dwServiceFlag;
             HRESULT hr = g_IconHandlers[i].pfnShutdown(this);
-            if (FAILED(hr))
-                return hr;
+            FAILED_UNEXPECTEDLY(hr);
```
Comment (HBelusca) [inline dll/shellext/stobject/csystray.cpp:146]: Is this just for diagnostics?

File: dll/shellext/stobject/csystray.cpp [hunk 3]
```diff
@@ -30,8 +38,7 @@ VOID CSysTray::GetServicesEnabled()
     /* Enable power, volume and hotplug by default */
     this->dwServicesEnabled = POWER_SERVICE_FLAG | VOLUME_SERVICE_FLAG | HOTPLUG_SERVICE_FLAG;
 
-    if (RegCreateKeyExW(HKEY_CURRENT_USER,
-                        L"Software\\Microsoft\\Windows\\CurrentVersion\\Applets\\SysTray",
+    if (RegCreateKeyExW(HKEY_CURRENT_USER, REGSTR_PATH_SYSTRAY,
```
Comment (HBelusca) [inline dll/shellext/stobject/csystray.cpp:41]:
```suggestion
    if (RegCreateKeyExW(HKEY_CURRENT_USER,
                        REGSTR_PATH_SYSTRAY,
```

File: dll/shellext/stobject/stobject.rc [hunk 4]
```diff
@@ -31,6 +31,18 @@ IDI_HOTPLUG_ERR ICON "resources/hotplug/0.ico"
 IDI_HOTPLUG_OK ICON  "resources/hotplug/1.ico"
 
 
+IDI_MOUSE_DISABLED ICON "resources/mouse/disabled.ico"
```
Comment (HBelusca) [inline dll/shellext/stobject/stobject.rc:34]:
```suggestion
IDI_MOUSE_DISABLED ICON "resources/mouse/disabled.ico"
```

**Review by HBelusca — APPROVED** [review]

File: dll/shellext/stobject/csystray.cpp (hunk 2 above)
Comment (learn-more) [reply dll/shellext/stobject/csystray.cpp:146]: yes

**Pull request merged**